Content.Server/CartridgeLoader/Cartridges/MessagesCartridgeSystem.cs

```diff
@@ -0,0 +1,304 @@
+using Content.Shared.CartridgeLoader;
+using Content.Shared.CartridgeLoader.Cartridges;
+using Content.Shared.PDA;
+using Robust.Shared.Map;
+using Content.Server.Radio.Components;
+using Content.Server.GameTicking;
+using Robust.Shared.Timing;
+using Content.Shared.Access.Components;
+using Content.Server.Radio.EntitySystems;
+using Content.Server.DeviceNetwork.Systems;
+using Content.Shared.DeviceNetwork;
+using Content.Server.Station.Systems;
+
+namespace Content.Server.CartridgeLoader.Cartridges;
+
+public sealed class MessagesCartridgeSystem : EntitySystem
+{
+    [Dependency] private readonly CartridgeLoaderSystem _cartridgeLoaderSystem = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly GameTicker _gameTicker = default!;
+    [Dependency] private readonly MessagesServerSystem _messagesServerSystem = default!;
+    [Dependency] private readonly DeviceNetworkSystem _deviceNetworkSystem = default!;
+    [Dependency] private readonly SingletonDeviceNetServerSystem _singletonServerSystem = default!;
+    [Dependency] private readonly StationSystem _stationSystem = default!;
+
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<MessagesCartridgeComponent, CartridgeMessageEvent>(OnUiMessage);
+        SubscribeLocalEvent<MessagesCartridgeComponent, CartridgeUiReadyEvent>(OnUiReady);
+        SubscribeLocalEvent<MessagesCartridgeComponent, DeviceNetworkPacketEvent>(OnPacketReceived);
+        SubscribeLocalEvent<MessagesCartridgeComponent, CartridgeActivatedEvent>(OnCartActivation);
+        SubscribeLocalEvent<MessagesCartridgeComponent, CartridgeDeactivatedEvent>(OnCartDeactivation);
+        SubscribeLocalEvent<MessagesCartridgeComponent, CartridgeAddedEvent>(OnCartInsertion);
+    }
+
+    /// <summary>
+    /// This gets called when the ui fragment needs to be updated for the first time after activating
+    /// </summary>
+    private void OnUiReady(EntityUid uid, MessagesCartridgeComponent component, CartridgeUiReadyEvent args)
+    {
+        var stationId = _stationSystem.GetOwningStation(uid);
+        if (stationId.HasValue)
+            _singletonServerSystem.TryGetActiveServerAddress<MessagesServerComponent>(stationId.Value, out _);
+        UpdateUiState(uid, args.Loader, component);
+    }
```

Comment:
This hasvalue is unnecessary just pass the nullable entityuid in.

Reply:
TryGetActiveServerAddress doesn't seem to take a nullable argument, so I'm not sure what you mean.